core/include/detray/geometry/propagation_surface.hpp

```diff
@@ -0,0 +1,101 @@
+
```

Comment:
Why do you separate surface and propagation_surface? Can it be in the same class?

Reply:
I did it so that the propagation math is not compiled in case someone wants to only use geometry and navigation, like Celeritas and the ray tracer

Reply:
Hmm I don't think we have to be that pragmatic :thinking:

Reply:
Ok, I agree it is ugly. I can write my own surface class in the ray tracer, since the class is not integrated into the detector

Reply:
No, I can't compile the math part, because there is no SoA implementation for it. But that's something for me to worry about

Reply:
I see. In that case I am ok with writing another surface class for ray tracer tentatively